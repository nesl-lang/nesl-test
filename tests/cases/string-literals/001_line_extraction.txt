R"""pv(simple)pv"""
R"""pv()pv"""
R"""pv(contains )pv""" in the middle)pv"""
R"""pv(has )pv""" and also R"""pv( patterns)pv"""
R"""pv()pv""" at the beginning)pv"""
R"""pv(ends with )pv""")pv"""
R"""pv(R"""pv(not actually nested)pv""")pv"""
R"""pv(multiple )pv""" and )pv""" and even )pv""" patterns)pv"""
prefix R"""pv(string with prefix)pv"""
R"""pv(string with suffix)pv""" suffix
  R"""pv(indented string)pv"""
    R"""pv(deeply indented)pv"""
key = R"""pv(assignment context)pv"""
  key = R"""pv(indented assignment)pv"""
- R"""pv(array element)pv"""
  - R"""pv(indented array element)pv"""
R"""pv(first)pv""" R"""pv(second)pv"""
R"""pv(one)pv""" R"""pv(two)pv""" R"""pv(three)pv"""
R"""pv(text)pv""")pv""")pv""")pv"""
R"""pv(unterminated string
R"""pv(good)pv""" extra content
R"""pv(valid)pv"""   
no string markers here
  R"""pv(missing close