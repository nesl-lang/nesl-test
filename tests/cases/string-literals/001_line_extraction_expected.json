[
  {"value": "simple"},
  {"value": ""},
  {"value": "contains )pv\"\"\" in the middle"},
  {"value": "has )pv\"\"\" and also R\"\"\"pv( patterns"},
  {"value": ")pv\"\"\" at the beginning"},
  {"value": "ends with )pv\"\"\""},
  {"value": "R\"\"\"pv(not actually nested)pv\"\"\""},
  {"value": "multiple )pv\"\"\" and )pv\"\"\" and even )pv\"\"\" patterns"},
  {"error": "invalid_string_start"},
  {"error": "content_after_string"},
  {"value": "indented string"},
  {"value": "deeply indented"},
  {"error": "invalid_string_start"},
  {"error": "invalid_string_start"},
  {"error": "invalid_string_start"},
  {"error": "invalid_string_start"},
  {"value": "first)pv\"\"\" R\"\"\"pv(second"},
  {"value": "one)pv\"\"\" R\"\"\"pv(two)pv\"\"\" R\"\"\"pv(three"},
  {"value": "text)pv\"\"\")pv\"\"\")pv\"\"\""},
  {"error": "string_unterminated"},
  {"error": "content_after_string"},
  {"value": "valid"},
  {"error": "string_not_found"},
  {"error": "string_unterminated"}
]